### ISSUE TITLE
Team to channel conversion error

### ISSUE THREAD

**Issue by sumitsrv** [RocketChat/Rocket.Chat#27526]
### Description:

I am trying to convert a team to a channel, but I receive an error: “team-does-not-exist”. I have noticed this error a few more times, but does not happen always. Furthermore, I tried both via the interface and the API. Attached are the screenshots of the interface with the error corresponding to the team I am trying to convert. Also, I have attached the respective logs from the server.

### Steps to reproduce:
Reproduction is difficult, given that there are several teams created that were easily converted to channel without any issue. The process to create the channel is as follows,
1. Create a team with three users (User 1, User 2, Admin). API used: /teams.create, type: 0
2. Call the API /teams.convertToChannel to convert the team to channel. The team name is used to perform this operation. (Also tried with team_id after the failure but it did not work).

### Expected behavior:

The team should be converted to a channel.

### Actual behavior:

I receive an error: "team-does-not-exist".

### Server Setup Information:

- Version of Rocket.Chat Server: 4.8.7 (updated to 5.4.0 after the issue)
- Operating System: Ubuntu 20.04
- Deployment Method: Snap
- Number of Running Instances: 1 
- DB Replicaset Oplog:  As per Snap installation
- NodeJS Version: As per Snap installation
- MongoDB Version: As per Snap installation

### Client Setup Information

- Desktop App or Browser Version: Primary - API call (Python), Firefox, Chrome
- Operating System: Ubuntu 20.04

### Relevant logs:
![Screenshot from 2022-12-09 18-18-46_2](https://user-images.githubusercontent.com/4637659/207301409-63f94fc6-ed99-4683-bbf9-c1a418d4c18c.png)

![Screenshot from 2022-12-09 18-58-22](https://user-images.githubusercontent.com/4637659/207301416-9c97585a-2e67-49e4-8a7e-f0b09d59809d.png)

**Comment by debdutdeb** [comment]
just to confirm happens on 4.8.7 or 5.4.0?

**Comment by sumitsrv** [comment]
Originally it occurred on 4.8.7. I updated the existing installation via
Snap to 5.4.0 to see if  issue persists. With this new version, I tried
doing the conversion from team to channel for the team it was failing for
but the response remained the same. I did not execute the entire process of
creating a new team and its conversion to channel.

**Comment by debdutdeb** [comment]
So it's happening everytime for that specific team?

**Comment by sumitsrv** [comment]
Yes, for the team that you can see in the screenshot added in the original
post.

If you check the logs, the first API call to teams.listRooomsOfUser itself
returns 400. As per my understanding, this call is made when the conversion
API is called. So, apparently, Rocket.chat is unable to find the respective
team for the user in the DB or something related to it is happening.

**Comment by debdutdeb** [comment]
Okk .. I'll see if I can reproduce it or not. 

Can I ask you to not remove that room for now?

**Comment by sumitsrv** [comment]
Sure. I will retain the set-up and the data as in the current state.
